opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java

```diff
@@ -595,12 +622,49 @@ public String getVersion() {
         return "2.0";
     }
 
-    public void setInputTransformer(InputTransformer inputTransformer) {
-        this.inputTransformer = inputTransformer;
+    private InputTransformer getInputTransformer(InputStream inputStream) {
+        XMLStreamReader xmlStreamReader = null;
+        try {
+            xmlStreamReader = xmlInputFactory.createXMLStreamReader(inputStream);
+            Bundle bundle = FrameworkUtil.getBundle(this.getClass());
+            if (bundle == null) {
+                // only happens during unit tests
+                return defaultInputTransformer;
+            }
+            BundleContext bundleContext = bundle.getBundleContext();
+            Collection<ServiceReference<InputTransformer>> serviceReferences = bundleContext
```

Comment:
We should be able to use a reference-list in blueprint and avoid using the FrameworkUtil and BundleContext directly.  Ideally we keep this a POJO that knows nothing about OSGi. That will help make this more testable too.